firmware/util/math/interpolation.h

```diff
@@ -46,7 +46,8 @@ int needInterpolationLogging(void);
  */
 template<typename kType>
 int findIndexMsgExt(const char *msg, const kType array[], int size, kType value) {
-	if (cisnan(value)) {
+	float fvalue = (float)value;
```

Comment:
If kType isn't a float, then it can't be NaN anyway.  It'd be better to do `if (std::is_floating_point_v<kType> && cisnan(value))`  That way, cisnan is never called if `kType` isn't a float